chai_lab/data/dataset/embeddings/esm.py

```diff
@@ -45,28 +51,82 @@ def esm_model(model_name: str, device):
     model.to("cpu")  # move model back to CPU when done
 
 
+token_map = {
+    "<cls>": 0,
+    "<pad>": 1,
+    "<eos>": 2,
+    "<unk>": 3,
+    "L": 4,
+    "A": 5,
+    "G": 6,
+    "V": 7,
+    "S": 8,
+    "E": 9,
+    "R": 10,
+    "T": 11,
+    "I": 12,
+    "D": 13,
+    "P": 14,
+    "K": 15,
+    "Q": 16,
+    "N": 17,
+    "F": 18,
+    "Y": 19,
+    "M": 20,
+    "H": 21,
+    "W": 22,
+    "C": 23,
+    "X": 24,
+    "B": 25,
+    "U": 26,
+    "Z": 27,
+    "O": 28,
+    ".": 29,
+    "-": 30,
+    "<null_1>": 31,
+    "<mask>": 32,
+}
+
+
+class DumbTokenizer:
+    def __init__(self, token_map: dict[str, int]):
+        self.token_map = token_map
+
+    def tokenize(self, text: str) -> list[int]:
+        tokens = []
+        i = 0
+        while i < len(text):
+            for token in self.token_map:
+                if text.startswith(token, i):
```

Comment:
Do we increment `text` anywhere like doing a `text[i:].startswith(...)`? I'm not seeing this but maybe I'm missing something.

Reply:
i is incremented, string not changed. We pass i in startswith

Reply:
Didn't realize that `startswith` took the start index as an argument - cool